frontend/src/lib/components/UserDataInput.svelte

```diff
@@ -1,117 +1,166 @@
+	import { onMount } from 'svelte';
+	import { _ } from 'svelte-i18n';
+
+	function convertQuestions(questionnaire: GetUserQuestionsResponse): GetUserQuestionsResponse {
+		// TODO: once questions come from the database we can do this with the questionnaire
+		// different algorithm that makes use of the component lookup table in '$lib/stores' is needed here then.
+		return questionnaire;
 	}
 
-	async function acceptData() {
-		const valid = validate();
-		if (valid) {
-			for (let i = 0; i < data.length; ++i) {
-				(userData as UserData)[data[i].props.name] = {};
-				(userData as UserData)[data[i].props.name]['value'] = data[i].value;
-				(userData as UserData)[data[i].props.name]['additionalValue'] = data[i].additionalValue;
+	function convertAnswers(questionnaire: GetUserQuestionsResponse | undefined): UserAnswerPublic[] {
+		// TODO: once questions come from the database we can do this with the actual questionnaire
+		// this version is temporary until the underlying datastructure has been changed to accommodate
+		// the backend models
+
+		return data.map((e, index) => {
+			console.log(
+				'condition: ',
+				e.additionalValue !== '',
+				e.additionalValue !== null,
+				e.value === e.textTrigger,
+				e.value,
+				e.additionalValue,
+				e.props.textTrigger
+			);
+			if (
+				e.additionalValue !== '' &&
+				e.additionalValue !== null &&
+				e.value === e.props.textTrigger
+			) {
+				console.log('with additional: ', e);
+				return {
+					question_id: index,
+					answer: String(e.additionalValue),
+					non_standard: true
+				} as UserAnswerPublic;
+			} else {
+				console.log('with normal', e);
+				return {
+					question_id: index,
+					answer: String(e.value),
+					non_standard: false
+				} as UserAnswerPublic;
 			}
+		});
+	}
 
-			if (userID) {
-				await users.update(userID, userData);
-			}
+	async function submitData() {
+		const answers = convertAnswers(questionnaire);
 
-			await users.save();
+		const response = await updateCurrentUserAnswers({
+			body: answers,
+			query: { session_id: userID }
+		});
 
-			buttons[0].disabled = true;
-			showAlert = false;
-			goto('/userLand/userLandingpage');
-		} else {
+		if (response.error) {
+			console.log('Error happened when sending user question answers: ', response.error);
+			alertMessage = $_('userData.alertMessageError');
 			showAlert = true;
+		} else {
+			console.log('successfully sent data to backend, yay!');
+			console.log('answers sent: ', answers);
+			console.log(
+				'original answers: ',
+				data.map((e) => {
+					return [e.value, e.additionalValue];
+				})
+			);
+
+			// disable all elements to make editing a conscious choice
+			for (let element of data) {
+				element.props.disabled = true;
+			}
+			dataIsCurrent = true;
+			data = [...data]; // TODO: forces a rerender. need a better way to do this
 		}
 	}
 
-	let userData: UserData;
-	let userID: string;
+	// README: is storing userID a problem?
```

Comment:
why do we need to store it?

Reply:
I managed to do things without it, this was a leftover from an earlier iteration...